src/query/procedure/module.cpp

```diff
@@ -560,6 +906,9 @@ bool PythonModule::Close() {
     py_module_ = py::Object(nullptr);
     return false;
   }
+
+  // Remove the cached bytecode if it's present
+  std::filesystem::remove_all(file_path_.parent_path() / "__pycache__");
```

Comment:
Just to be sure: `__pycache__` is only used when the interpreter loads a module and it is not used after that, right? It would be awful to delete a file while the interpreter is using it.

Reply:
These operations require the write lock on module registry (`Unload/LoadModules`), while the interpreter gets read lock on each procedure call so I would say it's safe.